src/controller/CHIPCluster.cpp

```diff
@@ -61,7 +62,12 @@ CHIP_ERROR ClusterBase::SendCommand(uint8_t seqNum, chip::System::PacketBufferHa
         mDevice->AddResponseHandler(seqNum, onSuccessCallback, onFailureCallback);
     }
 
-    err = mDevice->SendMessage(Protocols::TempZCL::MsgType::TempZCLRequest, std::move(payload));
+    if (onFailureCallback != nullptr || onFailureCallback != nullptr)
```

Comment:
do we need two checks for onFailureCallback here?

Reply:
Good catch!  First one should have been onSuccessCallback.